### PR TITLE
cmd/snap-recovery-chooser: tweaks

### PR DESCRIPTION
**Pull request by bboozzoo**
Follow ups after #8305 has landed.

### CODE REVIEW

File: cmd/snap-recovery-chooser/main.go [hunk 1]
```diff
@@ -129,29 +129,28 @@ func runUI(cmd *exec.Cmd, sys *ChooserSystems) (rsp *Response, err error) {
...
+		return err
 	}
-	return err
+	return nil
 }
 
-func chooser(cli *client.Client) error {
-	snappyTesting := os.Getenv("SNAPPY_TESTING_USE_STDOUT") != ""
+func chooser(cli *client.Client) (reboot bool, err error) {
+	snappyTesting := osutil.GetenvBool("SNAPPY_TESTING_USE_STDOUT", false)
```
Comment (pedronis) [inline cmd/snap-recovery-chooser/main.go:139]: this also ignores the marker, right?

s/SNAPPY_TESTING_USE_STDOUT/SNAPD_CHOOSER_TESTING_DIRECT/ ?

s/snappyTesting/testingMode/ ?

File: cmd/snap-recovery-chooser/main_test.go [hunk 2]
```diff
@@ -455,12 +476,89 @@ echo '{"label":"label","action":{"mode":"install","title":"reinstall"}}'
...
+	called := false
+	r := main.MockSyslogNew(func(_ syslog.Priority, _ string) (io.Writer, error) {
+		called = true
+		return nil, fmt.Errorf("no syslog")
+	})
+	defer r()
+	err = main.LoggerWithSyslogMaybe()
+	c.Assert(err, IsNil)
+	c.Check(called, Equals, true)
+	// this likely goes to stder
```
Comment (zyga) [inline cmd/snap-recovery-chooser/main_test.go:524]:
```suggestion
	// this likely goes to stderr
```

File: cmd/snap-recovery-chooser/main.go (hunk 1 above)
Comment (mvo5) [reply cmd/snap-recovery-chooser/main.go]: We could do something like https://paste.ubuntu.com/p/sXqWJkT7wp/ and check the outputForUI this way (or keeping in a separate test with something like https://paste.ubuntu.com/p/s86HQY8zfn/) - then we don't need this at all (FWIW, I don't like this special test mode as the change in the tests is not that bad IMHO to justify that we keep this in the prod code around). Unless I miss something of course?

Comment (bboozzoo) [reply cmd/snap-recovery-chooser/main.go]: I think we can drop this now. I took another look at the local testing, and this is not necessary anymore.

File: cmd/snap-recovery-chooser/main.go [hunk 3]
```diff
@@ -129,70 +128,94 @@ func runUI(cmd *exec.Cmd, sys *ChooserSystems) (rsp *Response, err error) {
 }
 
 func cleanupTriggerMarker() error {
-	err := os.Remove(defaultMarkerFile)
-	if os.IsNotExist(err) {
-		return nil
+	if err := os.Remove(defaultMarkerFile); err != nil && !os.IsNotExist(err) {
```
Comment (mvo5) [inline cmd/snap-recovery-chooser/main.go:131]: :heart:

File: cmd/snap-recovery-chooser/main.go [hunk 4]
```diff
@@ -129,70 +128,94 @@ func runUI(cmd *exec.Cmd, sys *ChooserSystems) (rsp *Response, err error) {
...
 
-	if err := chooser(client.New(nil)); err != nil {
+	reboot, err := chooser(client.New(nil))
+	if err != nil {
 		logger.Noticef("cannot run recovery chooser: %v", err)
 		fmt.Fprintf(Stderr, "%v\n", err)
 		os.Exit(1)
 	}
+	if reboot {
+		fmt.Fprintf(Stderr, "The system is rebooting...\n")
```
Comment (mvo5) [inline cmd/snap-recovery-chooser/main.go:219]: Nice

File: cmd/snap-recovery-chooser/main_test.go [hunk 5]
```diff
@@ -225,9 +237,11 @@ func (s *mockedClientCmdSuite) TestMainChooserWithTool(c *C) {
 	// sanity
 	c.Assert(s.markerFile, testutil.FilePresent)
 
-	mockCmd := testutil.MockCommand(c, "tool", `
+	capturedStdinPath := filepath.Join(c.MkDir(), "stdin")
+	mockCmd := testutil.MockCommand(c, "tool", fmt.Sprintf(`
+cat - > %s
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply cmd/snap-recovery-chooser/main_test.go:242]: @mvo5 proposed that, I'm only a messenger.

**Pull request merged**